routes/baseDBMethods.js

```diff
@@ -22,6 +24,35 @@ function handleDocResponse(ctx, doc) {
   }
 }
 
+function proccessDatesRangeParameters(ctx) {
+  const startDate = parseInt(ctx.query.startDate, 10) || null;
```

Comment:
I'm not sure that it's justified to use ` || null`, since if it's not an integer, `parseInt` returns `NaN`, which you can check with `!Number.isNaN(variable)` instead of `variable != null`.

Reply:
It will return NaN even if ctx.query.startDate is undefined?